src/components/tabset/tab/tab.component.ts

```diff
@@ -26,20 +28,52 @@ export class TabComponent {
     /** provide a custom class for the tab */
     @Input() customClass: string;
 
-    /** Emit when this tab is selected */
+    /** @deprecated Emit when this tab is selected - use activated output instead */
+        // tslint:disable-next-line
     @Output() select = new EventEmitter<void>();
 
-    /** Emit when this tab is deselected */
+    /** Emit when this tab is selected */
+    @Output() activated = new EventEmitter<void>();
+
+    /** @deprecated Emit when this tab is deselected - - use deactivated output instead */
     @Output() deselect = new EventEmitter<void>();
 
+    /** Emit when this tab is deselected */
+    @Output() deactivated = new EventEmitter<void>();
+
     /** Store a custom header templateRef */
     @ContentChild(TabHeadingDirective, { read: TemplateRef, static: false }) headingRef: TemplateRef<void>;
 
+    /** Unsubscribe from all subscriptions when component is destroyed */
+    private _onDestroy = new Subject<void>();
+
     constructor(
         private readonly _tabset: TabsetService,
         private readonly _changeDetector: ChangeDetectorRef
     ) { }
 
+    ngOnInit(): void {
+        // warn anyone using the select output that it is deprecated
+        if (this.select.observers.length > 0) {
+            console.warn('Tab Component - (select) output is deprecated use activated instead');
+        }
+
+        // warn anyone using the deselect output that it is deprecated
+        if (this.deselect.observers.length > 0) {
+            console.warn('Tab Component - (deselect) output is deprecated use deactivated instead');
+        }
+
+        // for backwards compatibility emit deprecated select event
+        this.activated.pipe(takeUntil(this._onDestroy)).subscribe(() => this.select.emit());
+
+        // for backwards compatibility emit deprecated deselect event
+        this.deactivated.pipe(takeUntil(this._onDestroy)).subscribe(() => this.deselect.emit());
+    }
+
+    ngOnDestroy(): void {
+        this._onDestroy.next();
```

Comment:
We need to complete the destroy obervable here as well.

```typescript
this._onDestroy.complete();
```

Reply:
Updated